R/style.R

```diff
@@ -1,7 +1,13 @@
 # Helpers --------------------------------------------------------------------
 
 ## anticipates usage where the `...` bits make up one line
+##
+## 'usethis.quiet' is an undocumented option; anticipated usage:
+##   * eliminate `capture_output()` calls in usethis tests
+##   * other packages, e.g., devtools can call usethis functions quietly
 cat_line <- function(...) {
+  quiet <- getOption("usethis.quiet", default = FALSE)
```

Comment:
Wonder if it would be worth promoting this to an argument (with the same default value), so you could call `cat_line(quiet = FALSE)`. e.g.

```r
cat_line <- function(..., quiet = getOption("usethis.quiet", default = FALSE)) {
```

Maybe only useful for internal testing in usethis, but I think it is a little more natural.

Reply:
For this to be productive internally, I'd need to add a `quiet` argument to `todo()` and `done()` and then this would happen in a few places:

Before:
```
if (!quiet) {
    done("Adding {collapse(value(new))} to {value(proj_rel_path(path))}")
}
```

After:
```
done("Adding {collapse(value(new))} to {value(proj_rel_path(path))}", quiet = quiet)
```

To be useful externally, I'd need to add `quiet` to user-facing `use_*()` functions.

But I see how that is a nicer definition of `cat_line()` itself, even if I don't do the other bits.

Reply:
And ... done.

Reply:
OK I did that, but one downside is that a future usethis developer could make a direct call to `cat_line()` and use `quiet = FALSE` argument and the "usethis.quiet = TRUE" option won't have the promised effect of silence. But presumably they'd either suffer immediately in tests or a downstream maintainer would raise a concern.